cpp/json_ffi/config.cc

```diff
@@ -142,6 +144,63 @@ std::optional<std::vector<Data>> Conversation::AsPrompt(std::string* err) {
           }
         }
         message += role_text;
+      } else if (item["type"] == "image_url") {
+        if (item.find("image_url") == item.end()) {
+          *err += "Content should have an image_url field";
+          return std::nullopt;
+        }
+        std::string image_url =
+            item.at("image_url");  // TODO(mlc-team): According to OpenAI API reference this
+                                   // should be a map, with a "url" key containing the URL, but
+                                   // we are just assuming this as the URL for now
+        std::string base64_image = image_url.substr(image_url.find(",") + 1);
+        std::optional<NDArray> image_data = LoadImageFromBase64(base64_image, err);
+        if (!image_data) {
+          return std::nullopt;
+        }
+
+        if (config.find("model_config") == config.end()) {
+          *err += "model_config is required in config";
+          return std::nullopt;
+        }
+        if (!config["model_config"].is<picojson::object>()) {
+          *err += "model_config should be an object";
+          return std::nullopt;
+        }
+        picojson::object model_config = config["model_config"].get<picojson::object>();
```

Comment:
Use explicit object. In this case, ideally we should do `const picojson::object& model_config`, otherwise it will trigger a deep copy of the object. Of course moving to explicit in-memory structure will avoid the issue. Let us do that